index.html

```diff
@@ -40,5 +61,361 @@ <h2>
+            <p>
+              The user agent MUST apply any provided <var>constraints</var> to the produced
+              media after permission has been granted.
+            </p>
+            <p>
+              In the case of audio, the user agent MAY present the end-user with an option to
+              include audio from the current viewport in the capture, if available. Like
+              {{MediaDevices/getDisplayMedia()}} with regards to audio+video, the user agent is
+              allowed to not return audio even if the audio constraint is present. If the user
+              agent knows no audio will be shared for the lifetime of the stream it MUST NOT
+              include an audio track in the resulting stream. The user agent MAY accept a
+              request for audio and video by only returning a video track in the resulting
+              stream, or it MAY accept the request by returning both an audio track and a video
+              track in the resulting stream. The user agent MUST reject audio-only requests.
+            </p>
+            <p>
+              Like {{MediaDevices/getDisplayMedia()}}, the {{PermissionState/"granted"}}
+              permission cannot be persisted.
+            </p>
+            <p>When the {{MediaDevices/getViewportMedia()}}
+            method is called, the user agent MUST run the following
+            steps:</p>
+            <ol>
+              <li>
+                <p>If the <a>current settings object</a>'s
+                [=environment settings object/cross-origin isolated capability=]
+                is false, return a promise <a>rejected</a>
+                with a {{DOMException}} object whose {{DOMException/name}}
+                attribute has the value {{SecurityError}}.</p>
+              </li>
+              <li>
+                If the [=relevant settings object=]'s
+                [=environment settings object/responsible document=]'s
+                [=top-level browsing context=]'s
+                <a href="https://wicg.github.io/document-policy/#required-document-policy">
+                required document policy</a> does not contain
+                `Require-Document-Policy: html-capture` (TODO: use correct algorithm),
+                return a promise <a>rejected</a>
+                with a {{DOMException}} object whose {{DOMException/name}}
+                attribute has the value {{SecurityError}}.</p>
+              </li>
+              <li>
+                <p>If the [=relevant global object=] of [=this=] does not have
+          [=transient activation=], return a promise <a>rejected</a>
+                with a {{DOMException}} object whose {{DOMException/name}}
+                attribute has the value {{InvalidStateError}}.</p>
+              </li>
+              <li>
+                <p>Let <var>constraints</var> be the method's first
+                argument.</p>
+              </li>
+              <li>
```

Comment:
instead of copying the content of screen capture, it would be better to make that algorithm re-usable there.

Reply:
The algorithms are different so that doesn't work:
```diff
- Prompt the user to choose a display device, for a PermissionDescriptor with its name set
- to "display-capture", resulting in a set of provided media.
+ Request permission to use viewport capture, for a PermissionDescriptor with its name set
+ to "viewport-capture", resulting in a set of provided media.

- The provided media MUST include precisely one video track.
+ The provided media MUST include precisely one video track, which MUST be a live-capture
+ of the browser display surface of the relevant settings object's responsible document's
+ top-level browsing context's viewport.

- The devices chosen MUST be the ones determined by the user. Once selected, the source of a
- MediaStreamTrack MUST NOT change, unless the user permits it through their interaction with
- the user agent.
+ The source of a MediaStreamTrack MUST NOT change.

- User agents are encouraged to warn users against sharing browser display devices as well as
- monitor display devices where browser windows are visible, or otherwise try to discourage
- their selection on the basis that these represent a significantly higher risk when shared.
+ 

- The provided media MUST include at most one audio track. This audio track MUST NOT be
- included if audio was not specified in requestedMediaTypes, or if it was specified as false.
+ The provided media MUST include at most one audio track, which, if provided, MUST be the
+ combined audio produced by the sum of documents that consist of the relevant settings object's
+ responsible document's top-level browsing context's active document, and all active documents
+ in nested browsing contexts of the relevant settings object's responsible document's top-level
+ browsing context.

- For the newly created MediaStreamTrack, the user agent MUST NOT capture the prompt that
- was shown to the user.
- 
- Information that is not currently rendered to the screen SHOULD be obscured in captures unless
- the application has been specifically authorized to access that content (e.g. through means such
- as elevated permissions).
+ 

- The user agent MUST NOT share audio without active user consent, for example if the capture of
- the video of a window is accompanied by capture of the audio of the entire system, including
- applications unrelated to that window.
+ The user agent MUST NOT share the audio other than audio emitted from the captured tab,
+ and MUST NOT share audio of the entire system.
```
There's also a precedent here in https://github.com/w3c/mediacapture-screen-share/pull/73.

Reply:
it still feels that an algorithm with parameters could handle both; but maybe that's an unneeded optimization; I worry about improvements being brought in one place and missed in the other.

Reply:
@eladalon1983 and I discussed this this morning and we both prefer to keep the algorithms separate.

We need to worry either way (whether improvements were meant for both or just one or the other), and WPT tests still need to be written for both, so we don't see it as a big savings for us as editors.